src/waitForElementToBeRemoved.js

```diff
@@ -2,8 +2,9 @@
 import waitFor, { type WaitForOptions } from './waitFor';
 import { ErrorWithStack } from './helpers/errors';
 
-const isRemoved = (result) =>
-  !result || (Array.isArray(result) && !result.length);
+function isRemoved<T>(result: T): boolean {
```

Comment:
I had to replace arrow function, because vscode has some problems with them when using generic types (syntax isn't highlighted properly)